lib/oli_web/pow/messages.ex

```diff
@@ -46,14 +46,26 @@ defmodule OliWeb.Pow.Messages do
   def pow_email_confirmation_email_confirmation_required(conn) do
     email = conn.params["user"]["email"]
 
-    if email do
-      """
-      To continue, check #{email} for a confirmation email.\n
-      If you don’t receive this email, check your Spam folder or verify that #{email} is correct.\n
-      You can close this tab if you received the email.
-      """
-    else
-      PowEmailConfirmation.Phoenix.Messages.email_confirmation_required(conn)
+    user =
+      case conn.assigns.action do
```

Comment:
Puting this in other function can be also good WDYT? 
something like:
  defp get_user_for_confirmation("/authoring/registration", email), do: Accounts.get_author_by_email(email)
  defp get_user_for_confirmation("/registration", email), do: Accounts.get_user_by(%{email: email})
  defp get_user_for_confirmation(_, email), do: nil